pkgs/development/libraries/v8/default.nix

```diff
@@ -152,12 +152,13 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ which ];
-  buildInputs = [ readline python icu patchelf ];
+  buildInputs = [ readline python icu patchelf ]
+  ++ stdenv.lib.optionals stdenv.isDarwin [ libtool ];
 
   NIX_CFLAGS_COMPILE = "-Wno-error=strict-overflow";
 
   buildFlags = [
-    "LINK=g++"
+    (if clangFlag == "1" then "LINK=clang++" else "LINK=g++")
```

Comment:
I think it's better practice to just do "LINK=c++", that should work in gcc and clang.